### PR TITLE
 #1 basic generalized script

### PR DESCRIPTION
**Pull request by sch00lb0y**
I have made generic script to generate doc. and tested this in my test repository http://github.com/sch00lb0y/tempCheck and here is the travis log https://travis-ci.org/sch00lb0y/tempCheck/builds/230224339 and the generated gh-pages https://sch00lb0y.github.io/tempCheck/ @mariobehling @niranjan94 
please review this 
Thanks

### CODE REVIEW

**Comment by imujjwal96** [comment]
There are some typos in the README file. Also I think the documentation can be a little more descriptive.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,48 @@
 # yaydoc
-Docs! Yay!
+
+[![Join the chat at https://gitter.im/fossasia/yaydoc](https://badges.gitter.im/fossasia/yaydoc.svg)](https://gitter.im/fossasia/yaydoc?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+> Docs! Yay!
+# Usage 
+- install the travis in your local machine
```
Comment (pri22296) [inline README.md]: remove `the` from `install the travis`

**Comment by niranjan94** [comment]
@sch00lb0y the first three commits are not supposed to be part of this PR. 

Only `basic generalized script` should be in this PR. Please fix that. 

And what issue is this linked to ?
Give a better commit message. 

And please read blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia/

File: README.md [hunk 2]
```diff
@@ -1,2 +1,48 @@
...
+```shell
+gem install travis
+```
+- setup add ssh to your github profile. In order to do please follow the instruction given by github(https://help.github.com/articles/connecting-to-github-with-ssh/)
+- go to folder where your ssh keys are generate. deafult folder(~/.ssh), login to your travis and execute the below command
```
Comment (pri22296) [inline README.md]: typo in `default`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,48 @@
...
+```shell
+travis login
+travis encrypt-file <filename deafault filename: idrsa>
```
Comment (pri22296) [inline README.md]: typo in `default`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,48 @@
...
+```
+- push the idrsa.enc file to the repository
+- set the following environmental variable to the travis
```
Comment (pri22296) [inline README.md]: replace `environmental` with `environment`

Comment (sch00lb0y) [inline README.md]: @pri22296 i'll improve the doc Thanks

**Comment by niranjan94** [comment]
@sch00lb0y resolve conflict. And we need only commit in the PR at the end.

**Comment by imujjwal96** [comment]
@sch00lb0y After the command `travis login` When I run the next command, i get 

`Can't figure out GitHub repo name. Ensure you're in the repo directory, or specify the repo name via the -r option (e.g. travis <command> -r <owner>/<repo>)`

I'm blindly following the documentation.
Also for the first command mention that `ruby` and `ruby-dev` must be installed in the system.

File: generate.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+cd gh-pages
+git rm -rf ./*
+cp -a ../_build/html/. ./
+git add -f .
+
+git commit -m "gh-pages push"
+git push -fq origin gh-pages
+cd ..
+cd ..
+chmod -R u+x temp
```
Comment (niranjan94) [inline generate.sh]: These last 4 lines are not needed. Travis would clean up after each run.

File: generate.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+#cp rep/doc temp/
+
+cd temp
+sphinx-quickstart -q -v $VERSION -a $AUTHOR -p $PROJECTNAME
+rm index.rst
+cd ..
+cp -a $DOCPATH. temp/
+cd temp
+make html
+git config --global user.email $EMAIL
```
Comment (niranjan94) [inline generate.sh]: Since this is gonna be an automated push by a machine, we do not want the commits linked to a specific person. Use 
```bash
git config --global user.name "Travis CI"
git config --global user.email "noreply+travis@fossasia.org"
```

File: generate.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+
+git clone --quiet --branch=gh-pages $GITURL gh-pages
+chmod -R u+x gh-pages
+cd gh-pages
+git rm -rf ./*
+cp -a ../_build/html/. ./
+git add -f .
+
+git commit -m "gh-pages push"
```
Comment (niranjan94) [inline generate.sh]: Let the commit message be something sensible like 
```
git commit -am "[Auto] Update Built Docs ($(date +%Y-%m-%d.%H:%M:%S))"
```

File: generate.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+git push -fq origin gh-pages
```
Comment (niranjan94) [inline generate.sh]: Any specific reason to do a force push ?

File: generate.sh [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+cp -a $DOCPATH. temp/
+cd temp
+make html
+git config --global user.email $EMAIL
+git config --global user.name $USERNAME
+
+
+
+git clone --quiet --branch=gh-pages $GITURL gh-pages
+chmod -R u+x gh-pages
```
Comment (niranjan94) [inline generate.sh]: Why is this needed ?

File: README.md [hunk 10]
```diff
@@ -3,3 +3,33 @@
 [![Join the chat at https://gitter.im/fossasia/yaydoc](https://badges.gitter.im/fossasia/yaydoc.svg)](https://gitter.im/fossasia/yaydoc?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
 > Docs! Yay!
+
+# Usage
+- generate github personal token by following this guide (https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+- set the following environment variable in the travis       
+     1) AUTHOR      (author of the repository)
+     2) DOCPATH     (path of the documentation)
+     3) EMAIL       (email of the author)
```
Comment (niranjan94) [inline README.md]: Email is not needed since we'll not be using it anywhere now.

File: README.md [hunk 11]
```diff
@@ -3,3 +3,33 @@
...
+     4) GITURL      (ssh url not https)
```
Comment (niranjan94) [inline README.md]: Change to `https url not ssh`

File: generate.sh (hunk 7 above)
Comment (sch00lb0y) [inline generate.sh]: i have changed Thanks

File: generate.sh (hunk 9 above)
Comment (sch00lb0y) [inline generate.sh]: when i tried removing folder without permission , error showing that no access to remove .git folder in gh-pages

File: generate.sh (hunk 6 above)
Comment (sch00lb0y) [inline generate.sh]: we can use this script on our yaydoc also

Comment (niranjan94) [inline generate.sh]: But the code will be run by a CI. And all fossasia projects use Travis.

Comment (niranjan94) [inline generate.sh]: Or keep it generic
```
git config --global user.name "Bot"
git config --global user.email "noreply+bot@fossasia.org"
```

File: generate.sh (hunk 9 above)
Comment (niranjan94) [inline generate.sh]: Weird. You tried it locally or on travis ?

File: generate.sh (hunk 5 above)
Comment (niranjan94) [inline generate.sh]: Hmmm ... Yep. Did think of that .. :) Cool !

File: generate.sh (hunk 6 above)
Comment (sch00lb0y) [inline generate.sh]: I thought it'll be generic. I'll change now

File: generate.sh (hunk 9 above)
Comment (sch00lb0y) [inline generate.sh]: Travis. Once I'll check again and I'll tell you

File: generate.sh [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+cd gh-pages
+git rm -rf ./*
+cp -a ../_build/html/. ./
+git add -f .
+
+git commit -m "gh-pages push"
+git commit -am "[Auto] Update Built Docs ($(date +%Y-%m-%d.%H:%M:%S))"
```
Comment (pri22296) [inline generate.sh]: where is the `git push` statement?

File: generate.sh (hunk 7 above)
Comment (niranjan94) [inline generate.sh]: @sch00lb0y why is this still there `git commit -m "gh-pages push"`

File: generate.sh (hunk 12 above)
Comment (niranjan94) [inline generate.sh]: And why are you amending the previous commit ? Let's have a history of changes.

File: generate.sh [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+git clone --quiet --branch=gh-pages $GITURL gh-pages
+#chmod -R u+x gh-pages
+cd gh-pages
+git rm -rf ./*
+cp -a ../_build/html/. ./
+git add -f .
+
+
+git commit -m "[Auto] Update Built Docs ($(date +%Y-%m-%d.%H:%M:%S))"
+git push -f gh-pages
```
Comment (niranjan94) [inline generate.sh]: Why is force push needed ?

File: generate.sh [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+cd ..
```
Comment (niranjan94) [inline generate.sh]: Can you combine both the `cd ..` into `cd ../../`

File: generate.sh [hunk 15]
```diff
@@ -23,7 +23,7 @@ git add -f .
 
 
 git commit -m "[Auto] Update Built Docs ($(date +%Y-%m-%d.%H:%M:%S))"
-git push gh-pages
+git push -f gh-pages
```
Comment (pri22296) [inline generate.sh]: It should be `git push -f origin gh-pages`

File: generate.sh (hunk 13 above)
Comment (niranjan94) [inline generate.sh]: Just a push should do.

File: generate.sh (hunk 6 above)
Comment (sch00lb0y) [inline generate.sh]: i just kept generic only

File: README.md (hunk 10 above)
Comment (sch00lb0y) [inline README.md]: for pushing we'll be needing

File: generate.sh (hunk 6 above)
Comment (niranjan94) [inline generate.sh]: By generic, I mean not a specific CI name.
```
git config --global user.name "Bot"
git config --global user.email "noreply+bot@example.com"
```

It needs to be known it's an automated commit and it should not be linked to any person.

File: README.md (hunk 10 above)
Comment (niranjan94) [inline README.md]: @sch00lb0y where is this used ?

File: generate.sh (hunk 6 above)
Comment (sch00lb0y) [inline generate.sh]: @niranjan94 okay i got it thanks

**Comment by niranjan94** [comment]
@sch00lb0y what is happening here ?

**Comment by sch00lb0y** [comment]
@niranjan94  when i'm resolving conflict i accidentally deleted the branch

**Comment by sch00lb0y** [comment]
@niranjan94 i'm so sorry i was not aware

**Comment by niranjan94** [comment]
@sch00lb0y please ensure this doesn't happen again. Now, we just have a new PR without any of of the conversation history.

**Comment by sch00lb0y** [comment]
@niranjan94 here after i won't happen. i'm sorry
